### ISSUE TITLE
TimeZone: Do not show a tooltip that just repeats the visible text

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#56211]
### Description

The TimeZone component uses a Tooltip to show an expansion of the UTC (Coordinated Universal Time) value.

There is a lot to say about using tooltips for descriptions, as the original goal of tooltips was to only show the accessible name of icon-only buttons. However, let's postpone this and just focus on the specific TimeZone implementation, for now.

It makes sense to expand the visible UTC when the additional information is actually an expansion. Screenshot:

However, it is not useful to show a tooltip when it just repeats the visible text. Screenshot:

The expansion text varies depending on the actual 'Timezone' setting in the admin, under Settings > General Settings > Timezone. When the value of the expansion is the same of the visible text, the  component should not use  Tooltip.

Worth also noting that now that [the Tooltip component has been refactored using ariakit](https://github.com/WordPress/gutenberg/pull/48440), it adds an `aria-describedby` to the element that uses the tooltip so that the element will have both the accessible name and the accessible description using the same text:

![Screenshot 2023-11-13 at 12 27 59](https://github.com/WordPress/gutenberg/assets/1682452/6b13261d-6f87-4d4d-9653-76df91bc5a37)

Testing with Safari + VoiceOver, it appears they are smart enough to avoid a double announcement but I'm pretty sure other screen readers will announce both the name and the description, which is redundant and confusing.

### Step-by-step reproduction instructions

- Go to the WordPress admin, Settings > General Settings > Timezone.
- Set the Timezone value to a _city_ value other than your city time zone e.g. Paris or New York.
- Save.
- Go to the post editor, edit a post or create a new one.
- Click the 'Publish' value in the settings panel.
- Observe the timezone value is an abbreviation, in my case is `EST` because I set it to New York.
- Hover with your mouse the timezone value, or move focus to it.
- Observe a Tooltip appears with the expanded value, in my case it's: `(EST) America/New York`.
- Go to the WordPress admin, Settings > General Settings > Timezone.
- Set the Timezone value to an _UTC value_ other than your one e.g. UTC+3.
- Save.
- Repeat the steps above.
- Observe the visible text is `UTC+3`.
- Observe the tooltip text is `UTC+3`.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Worth also noting that when the setting is something like `UTC+3:30`, the value shown in the UI is incorrect and it's `UTC+3.5`. Screenshot:

![Screenshot 2023-11-16 at 16 20 25](https://github.com/WordPress/gutenberg/assets/1682452/b4e52b06-1e26-49eb-b46f-c3f14ee6da21)

Edit:
It appears that in WordPress core there is some code to manipulate an _offset value_ like `3.5` and make it a _offset name_ for displaying purposes like `3:30`, see here:

https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/functions.php/#L6701-L6702

This part is missing in Gutenberg. it should be added because there is a clear need to have a correct offset _name_ for displaying purposes. Not sure whether it's best to add this directly in the `@wordpress/date` or just fix it here for displaying purposes.

**Comment by afercia** [comment]
Also noting that longer displayed timezone values may break the layout of the AM/PM buttons. Screenshot:

![Screenshot 2023-11-16 at 17 00 41](https://github.com/WordPress/gutenberg/assets/1682452/f15ff56d-1560-4e1c-a61d-f35a780e4c92)
